### DISCUSSION TITLE
CPP API to bulk load streaming data (I know this is OLAP)

### DISCUSSION THREAD

**Discussion by rvishureddy** [duckdb/duckdb#7949] [Q&A]
I get a stream of signal data from devices and the data is in stored in vectors. Is there a way i can insert this data into the duckdb using C++ API. 
All i could see is appender but that is row by row. Is there anything to bulk load.

**Comment by Tishj** [comment]
The Appender is used for bulk insertion
As documented here: https://duckdb.org/docs/data/appender

It's row-by-row but the insertions are cached first
